### PR TITLE
Support "lang" parameter for Koncorde filters

### PR DESCRIPTION
**Pull request by Yoann-Abbes**
## What does this PR do?

Since Kuzzle 2.8.0 supports the "lang" parameter to handle Koncorde filters, we need to add it in SDKs for the API methods who accept it. 

https://github.com/kuzzleio/sdk-javascript/issues/570

### How should this be manually tested?

  - Step 1 :
  - Step 2 :
  - Step 3 :  
  ...

### Boyscout

Update ancient links

### CODE REVIEW

File: doc/7/controllers/auth/search-api-keys/index.md [hunk 1]
```diff
@@ -30,6 +30,15 @@ searchApiKeys([query], [options]);
 
 The search query to apply to API keys content, using [ElasticSearch Query DSL](https://www.elastic.co/guide/en/elasticsearch/reference/7.3/query-dsl.html) syntax.
 
+<SinceBadge version="change-me"/>
```
Comment (Aschen) [inline doc/7/controllers/auth/search-api-keys/index.md:33]:
```suggestion
<SinceBadge version="auto-version"/>
```

Comment (Aschen) [reply doc/7/controllers/auth/search-api-keys/index.md:33]: Other badges must be updated as well

**Review by Aschen — CHANGES_REQUESTED** [review]
Can you add an example in the existing snippet for `document:search` at least ?
To ensure it's working with search result `next` method but also to show the feature to users

**Review by Njuelle — APPROVED** [review]

**Review by rolljee — APPROVED** [review]

File: doc/7/controllers/document/delete-by-query/snippets/delete-by-query-koncorde.js [hunk 2]
```diff
@@ -0,0 +1,14 @@
...
+      query: {
+        equals: { capacity: 7 }
+      }
+    }, { lang: 'koncorde' }
+  );
+
+  console.log(`Successfully deleted ${deleted.length} documents`);
+} catch (error) {
+  console.error(error.message);
+}
```
Comment (rolljee) [inline doc/7/controllers/document/delete-by-query/snippets/delete-by-query-koncorde.js:14]: missing newline

File: doc/7/controllers/document/update-by-query/snippets/update-by-query-es.test.yml [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+    for i in  1 2 3 4 5; do
+      curl --fail -H "Content-type: application/json" -d '{"capacity": 7}' kuzzle:7512/nyc-open-data/yellow-taxi/_create
+    done
+    curl -XPOST kuzzle:7512/nyc-open-data/yellow-taxi/_refresh
+  after:
+    - curl -XDELETE kuzzle:7512/nyc-open-data
+template: print-result-successes
+expected:
+  - "{ _id: 'document_1', _version: 2, status: 200 }"
+  - "{ _id: 'document_2', _version: 2, status: 200 }"
```
Comment (rolljee) [inline doc/7/controllers/document/update-by-query/snippets/update-by-query-es.test.yml:20]: missing newline

**Review by rolljee — CHANGES_REQUESTED** [review]

**Review by rolljee — APPROVED** [review]

File: doc/7/controllers/auth/search-api-keys/index.md [hunk 4]
```diff
@@ -60,4 +70,10 @@ Each object of the `hits` array has the following properties:
 
 ## Usage
 
-<<< ./snippets/search-api-keys.js
+With the [ElasticSearch Query DSL](https://www.elastic.co/guide/en/elasticsearch/reference/7.3/query-dsl.html) syntax.
```
Comment (Aschen) [inline doc/7/controllers/auth/search-api-keys/index.md:73]:
```suggestion
With the [ElasticSearch Query DSL](https://www.elastic.co/guide/en/elasticsearch/reference/7.4/query-dsl.html) syntax.
```

File: doc/7/controllers/auth/search-api-keys/snippets/search-api-keys-koncorde.js [hunk 5]
```diff
@@ -0,0 +1,66 @@
...
+  /*
+  {
+    "total": 2,
+    "hits": [
+      {
+        "_id": "znEwbG8BJASM_0-bWU-q",
+        "_source": {
+          "description": "LoRa permanent API key",
+          "userId": "jared.doe",
+          "expiresAt": 31557600000,
```
Comment (Aschen) [inline doc/7/controllers/auth/search-api-keys/snippets/search-api-keys-koncorde.js:46]: Missing from original doc
```suggestion
          "expiresAt": 31557600000,
          "fingerprint": "4ee98cb8c614e99213e7695f822e42325d86c93cfaf39cb40e860939e784c8e6",
```

File: doc/7/controllers/auth/search-api-keys/snippets/search-api-keys-koncorde.js [hunk 6]
```diff
@@ -0,0 +1,66 @@
...
+          "ttl": 420000
+        }
+      },
+      {
+        "_id": "zXEwbG8BJASM_0-bWU-q",
+        "_source": {
+          "description": "LoRa 1 year API key",
+          "userId": "jared.doe",
+          "expiresAt": 31557600000,
```
Comment (Aschen) [inline doc/7/controllers/auth/search-api-keys/snippets/search-api-keys-koncorde.js:55]: Missing from original doc
```suggestion
          "expiresAt": 31557600000,
          "fingerprint": "f822e42325d86c93cfaf39cb40e860939e784c8e64ee98cb8c614e99213e7695",
```

File: doc/7/controllers/document/delete-by-query/index.md [hunk 7]
```diff
@@ -34,11 +43,19 @@ Additional query options
...
+| `lang`     | <pre>string</pre>               | Specify the query language to use. By default, it's `elasticsearch` but `koncorde` can also be used. <SinceBadge version="auto-version"/> |
 
 ## Resolves
 
 Resolves to an array of strings containing the deleted document ids.
 
 ## Usage
 
-<<< ./snippets/delete-by-query.js
+With the [ElasticSearch Query DSL](https://www.elastic.co/guide/en/elasticsearch/reference/7.3/query-dsl.html) syntax.
```
Comment (Aschen) [inline doc/7/controllers/document/delete-by-query/index.md:55]:
```suggestion
With the [ElasticSearch Query DSL](https://www.elastic.co/guide/en/elasticsearch/reference/7.4/query-dsl.html) syntax.
```

File: doc/7/controllers/document/search/index.md [hunk 8]
```diff
@@ -73,4 +82,10 @@ Resolves to a [SearchResult](/sdk/js/7/core-classes/search-result) object.
 
 ## Usage
 
-<<< ./snippets/search.js
+With the [ElasticSearch Query DSL](https://www.elastic.co/guide/en/elasticsearch/reference/7.3/query-dsl.html) syntax.
```
Comment (Aschen) [inline doc/7/controllers/document/search/index.md:85]:
```suggestion
With the [ElasticSearch Query DSL](https://www.elastic.co/guide/en/elasticsearch/reference/7.4/query-dsl.html) syntax.
```

File: doc/7/controllers/document/update-by-query/index.md [hunk 9]
```diff
@@ -61,4 +70,10 @@ Each errored document is an object of the `errors` array with the following prop
 
 ## Usage
 
-<<< ./snippets/update-by-query.js
\ No newline at end of file
+With the [ElasticSearch Query DSL](https://www.elastic.co/guide/en/elasticsearch/reference/7.3/query-dsl.html) syntax.
```
Comment (Aschen) [inline doc/7/controllers/document/update-by-query/index.md:73]:
```suggestion
With the [ElasticSearch Query DSL](https://www.elastic.co/guide/en/elasticsearch/reference/7.4/query-dsl.html) syntax.
```

File: doc/7/controllers/security/search-api-keys/index.md [hunk 10]
```diff
@@ -61,4 +71,10 @@ Each object of the `hits` array has the following properties:
 
 ## Usage
 
-<<< ./snippets/search-api-keys.js
+With the [ElasticSearch Query DSL](https://www.elastic.co/guide/en/elasticsearch/reference/7.3/query-dsl.html) syntax.
```
Comment (Aschen) [inline doc/7/controllers/security/search-api-keys/index.md:74]:
```suggestion
With the [ElasticSearch Query DSL](https://www.elastic.co/guide/en/elasticsearch/reference/7.4/query-dsl.html) syntax.
```

File: doc/7/controllers/security/search-users/index.md [hunk 11]
```diff
@@ -24,7 +24,16 @@ searchUsers([query], [options]);
 
 ### query
 
-The search query to apply to users content, using [ElasticSearch Query DSL](https://www.elastic.co/guide/en/elasticsearch/reference/7.3/query-dsl.html) syntax.
+The search query to apply to users content, using [ElasticSearch Query DSL](https://www.elastic.co/guide/en/elasticsearch/reference/7.3/query-dsl.html) or the [Koncorde Filters DSL](/core/2/api/koncorde-filters-syntax) syntax.
```
Comment (Aschen) [inline doc/7/controllers/security/search-users/index.md:27]:
```suggestion
The search query to apply to users content, using [ElasticSearch Query DSL](https://www.elastic.co/guide/en/elasticsearch/reference/7.4/query-dsl.html) or the [Koncorde Filters DSL](/core/2/api/koncorde-filters-syntax) syntax.
```

File: doc/7/controllers/security/search-users/index.md [hunk 12]
```diff
@@ -36,11 +45,18 @@ If left empty, the result will return all available users.
...
 
 ## Resolves
 
-A [`SearchResult`](sdk/js/6/core-classes/search-result) object containing the retrieved [`User`](/sdk/js/7/core-classes/user) objects.
+A [`SearchResult`](sdk/js/7/core-classes/search-result) object containing the retrieved [`User`](/sdk/js/7/core-classes/user) objects.
 
 ## Usage
 
-<<< ./snippets/search-users.js
+With the [ElasticSearch Query DSL](https://www.elastic.co/guide/en/elasticsearch/reference/7.3/query-dsl.html) syntax.
```
Comment (Aschen) [inline doc/7/controllers/security/search-users/index.md:56]:
```suggestion
With the [ElasticSearch Query DSL](https://www.elastic.co/guide/en/elasticsearch/reference/7.4/query-dsl.html) syntax.
```

File: src/controllers/Auth.ts [hunk 13]
```diff
@@ -125,7 +125,7 @@ export class AuthController extends BaseController {
    */
   searchApiKeys(
     query: JSONObject = {},
-    options: { from?: number, size?: number } = {}
+    options: { from?: number, size?: number, lang?: string } = {}
```
Comment (Aschen) [inline src/controllers/Auth.ts:128]: Missing parameter documentation in the method description

File: src/controllers/Document.ts [hunk 14]
```diff
@@ -737,7 +741,7 @@ export class DocumentController extends BaseController {
     collection: string,
     query: JSONObject,
     changes: JSONObject,
-    options: { refresh?: 'wait_for', source?: boolean } = {}
+    options: { refresh?: 'wait_for', source?: boolean, lang?: string} = {}
```
Comment (Aschen) [inline src/controllers/Document.ts:744]: Nit
```suggestion
    options: { refresh?: 'wait_for', source?: boolean, lang?: string } = {}
```

File: src/controllers/Document.ts [hunk 15]
```diff
@@ -152,20 +152,22 @@ export class DocumentController extends BaseController {
...
+   *    - `lang` Query syntax. Can be 'elasticsearch' or 'koncorde'
    *
    * @returns The deleted documents IDs
    */
   deleteByQuery(
     index: string,
     collection: string,
     query: JSONObject = {},
-    options: { queuable?: boolean, refresh?: 'wait_for' } = {}
+    options: { queuable?: boolean, refresh?: string, lang?: string} = {}
```
Comment (Aschen) [inline src/controllers/Document.ts:163]: Nitpicking
```suggestion
    options: { queuable?: boolean, refresh?: string, lang?: string } = {}
```

**Review by Aschen — CHANGES_REQUESTED** [review]

**Review by Aschen — APPROVED** [review]

**Review by Njuelle — APPROVED** [review]

**Pull request merged**